feat: Editorial review: Fenced Frames docs (#30874)
mdn/content: files/en-us/web/api/fenced_frame_api/communication_with_embedded_frames/index.md

---

> title: Communication with embeded frames

Reviewer: ```suggestion
title: Communication with embedded frames
```

Author: Fixed, ta!

slug: Web/API/Fenced_frame_API/Communication_with_embedded_frames
page-type: guide
---

{{DefaultAPISidebar("Fenced Frame API")}}

This article provides information on how communication differs between an embedder and content embedded inside different types of frame (i.e. an {{htmlelement("iframe")}} and a {{htmlelement("fencedframe")}}), and how passed data can be stored.

## How to communicate between the embedder and an `<iframe>`

> ![Diagram illustrating the difference between local storage and shared storage and communication with an iframe, as explained below](iframe-storage-communication.png)

Reviewer: Local storage here looks like a DB, maybe later in the prose we can talk about client-side shared storage so that it's clear where this data is.

Author: Yup, that's the idea. I've included a mention of "client-side" in the prose, to hint more strongly at this.

When the third-party code is embedded in an `<iframe>`, the `<iframe>` and the embedder can send messages freely to each other to request data to be written into their [shared storage](/en-US/docs/Web/API/Shared_Storage_API). The embedder can send a request to that `<iframe>` to write data into its own third-party storage with a cross-document communication channel using {{domxref("Window.postMessage()")}}. The third party can also send `postMessage()` requests to the embedder.

> From the `<iframe>`, you can listen to a [`message`](/en-US/docs/Web/API/Window/message_event) event that comes from the embedder. When the embedder dispatches a message to the `<iframe>` using `postMessage()`, the `<iframe>` can take that data and store it in its own shared storage. Conversely, the `<iframe>` can dispatch a message that the embedder can listen to, and respond by writing data into its shared storage.

Reviewer: Can we say "client-side shared storage" or "shared storage on the client" - maybe even in both cases so it's clear. What do you reckon?

Author: Sure; done.

## How to communicate between the embedder and a `<fencedframe>`

> Sometime in the future, fenced frames will likely become more common as they will be used for use cases such as displaying targeted ads selected via the [Protected Audience API](https://developer.chrome.com/en/docs/privacy-sandbox/fledge/) and {{domxref("WindowSharedStorage.selectURL()")}}. Communicating between `<fencedframe>`s and other pages outside the `<fencedframe>` on the page is intentionally limited, but one method of communication between the embedder and shared storage worklets does exist — {{domxref("FencedFrameConfig.setSharedStorageContext()")}}.

Reviewer: Shall we make it a bit more future-proof?

```suggestion
Fenced frames are intended to be used for cases such as displaying targeted ads selected via the [Protected Audience API](https://developer.chrome.com/en/docs/privacy-sandbox/fledge/) and {{domxref("WindowSharedStorage.selectURL()")}}. Communicating between `<fencedframe>`s and other pages outside the `<fencedframe>` on the page is intentionally limited, but one method of communication between the embedder and shared storage worklets does exist — {{domxref("FencedFrameConfig.setSharedStorageContext()")}}.
```

Author: Yup, I like this. Updated!
